rocky/reports/forms.py

```diff
@@ -28,3 +30,97 @@ def __init__(self, report_types: set[Report], *args, **kwargs):
         super().__init__(*args, **kwargs)
         report_types_choices = ((report_type.id, report_type.name) for report_type in report_types)
         self.fields["report_type"].choices = report_types_choices
+
+
+class ReportScheduleForm(BaseRockyForm):
+    start_date = forms.DateField(
+        label=_("Start date"),
+        widget=DateInput(format="%Y-%m-%d", attrs={"form": "generate_report"}),
+        initial=lambda: datetime.now(tz=timezone.utc).date(),
+        required=False,
+    )
+
+    recurrence = forms.ChoiceField(
+        label=_("Recurrence"),
+        required=False,
+        widget=forms.Select(
+            attrs={"form": "generate_report"},
+        ),
+        choices=[
+            ("no_repeat", _("Does not repeat")),
+            ("daily", _("Daily")),
+            ("weekly", _("Weekly")),
+            ("monthly", _("Monthly")),
+            ("yearly", _("Yearly")),
+        ],
+    )
+
+    def is_valid_day_for_monthly_recurrence(self, start_date: date, recurrence: str) -> bool:
+        day = start_date.day
+        if recurrence == "monthly":
+            if day == 29:
+                self.add_error("start_date", _("Warning: Recurrence is set in February only for leap years."))
+                return False
+            if day == 30:
+                self.add_error("start_date", _("Warning: Recurrence is not set for February."))
+                return False
+            if day == 31:
+                self.add_error("start_date", _("Warning: Recurrence will skip months that does not have 31 days."))
+                return False
+        return True
```

Comment:
You could perhaps solve this problem by setting the cron expression to `0 0 28-31 * *`  when the start date is in the  28-31 range